test/template.test.js

```diff
@@ -156,73 +157,194 @@ test('[template.read] S3 JSON', function(assert) {
+    var name = questions[0];
+    assert.equal(name.default, 'Chuck', 'overriden default for Name');
+    assert.equal(name.message, 'Somebody', 'overriden message for Name');
+    name.async = function() {
+      return function(err, encrypted) {
+        assert.ok(encrypted, 'filter success for Name');
+        assert.equal(encrypted, 'Ham', 'passes through non-secret parameters');
+        next();
+      };
+    };
+    name.filter('Ham');
+  });
+
+  q.defer(function(next) {
+    var handedness = questions[2];
+    assert.deepEqual(handedness.choices, ['top', 'bottom'], 'overriden choices for Handedness');
+    next();
+  });
+
 
-  var handedness = questions[2];
-  assert.deepEqual(handedness.choices, ['top', 'bottom'], 'overriden choices for Handedness');
+  q.defer(function(next) {
+    var password = questions[5];
+    password.async = function() {
+      return function(err, encrypted) {
+        assert.ifError(err, 'encryption doesn\'t cause errors');
+        assert.equal(encrypted.slice(0, 7), 'secure:', 'encrypted var starts with secure:');
+        assert.equal((new Buffer(encrypted.slice(7), 'base64')).toString('utf8'), 'hibbities', 'decrypts correctly');
+        next();
+      };
+    };
+    password.filter('hibbities');
+  });
+
+  q.defer(function(next) {
+    var password = questions[5];
+    password.async = function() {
+      return function(err, encrypted) {
+        assert.ifError(err, 'encryption doesn\'t cause errors');
+        assert.equal(encrypted, 'secure:neverchange', 'passes over secure vars');
+        next();
+      };
+    };
+    password.filter('secure:neverchange');
+  });
+
+  q.awaitAll(function(err) {
+    AWS.restore('KMS');
+    assert.end(err);
+  });
+});
+
+test('[template.questions] defaults kms key to correct default', function(assert) {
+  AWS.mock('KMS', 'encrypt', function(params, callback) {
+    assert.equal(params.KeyId, 'alias/cloudformation', 'used default keyId');
+    assert.ok(params.Plaintext);
+    return callback(null, { CiphertextBlob: new Buffer(params.Plaintext) });
+  });
+
+  var questions = template.questions(expected, { kms: true });
 
-  assert.end();
+  var password = questions[5];
+  password.async = function() {
+    return function(err, encrypted) {
+      assert.ifError(err, 'encryption doesn\'t cause errors');
+      assert.equal(encrypted.slice(0, 7), 'secure:', 'encrypted var starts with secure:');
+      assert.equal((new Buffer(encrypted.slice(7), 'base64')).toString('utf8'), 'hibbities', 'decrypts correctly');
+      AWS.restore('KMS');
+      assert.end();
+    };
+  };
+  password.filter('hibbities');
+});
+
+test('[template.questions] handles failure during kms encryption', function(assert) {
+  AWS.mock('KMS', 'encrypt', function(params, callback) {
+    return callback(new Error('Bad encryption error'));
+  });
+
+  var questions = template.questions(expected, { kms: true });
+
+  var password = questions[5];
+  password.async = function() {
```

Comment:
By mocking like this on the main inquirer branch your tests are presenting a false version reality, right? This will only happen right now if you pin inquirer to your PR branch.